### DISCUSSION TITLE
License clarification.

### DISCUSSION THREAD

**Discussion by mionir** [minio/minio#16636] [Q&A] [answered]
Hi, the agpl v3 is quite "open" in terms of what it wants to do, but its also clear in what modification is, i intend on running minio as a backend storage for a few apis, that currently store files directly on disk, i dont intend on modifying minio in any way as it already checks most if not all marks needed for this.

My question is, no user will have direct access to minio and i will also be running on the already provided docker, in this case, to be compliant with the license, just a link back to minio repo saying "we use this awsome project" would be enough? or i would need to opensource all the apis that interact with it?

**Comment by harshavardhana** [comment, marked as answer]
Please consult your legal counsel on what the license entails. We can't give you legal feedback.

**Comment by kamikazechaser** [comment]
**Not Legal Advice.**

>  i dont intend on modifying minio in any way 

Under section 13 of the license, you don't need to provide source to the end user if it is unmodified. (The assumption is that MINIO already provides a link to this repo/source code in any form from the unmodified software)

> no user will have direct access to minio 

See whether you fit under the definition of "convey". In any case you don;t need to have the source public from my first point.
